server/tests/users.spec.js

```diff
@@ -0,0 +1,383 @@
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const app = require('../../app');
+const usersHelper = require('./helpers/userHelper');
+
+const dummyUser = usersHelper.usersHelper;
+
+const should = chai.should();
+chai.use(chaiHttp);
+
+describe('Users', () => {
+  let userToken;
+  let adminToken;
+  before((done) => {
+    chai.request(app)
+    .post('/api/users/login')
+    .send({
+      email: dummyUser[0].email,
+      password: dummyUser[0].password
+    })
+    .end((err, res) => {
+      userToken = res.body.Token;
+    });
+    chai.request(app)
+    .post('/api/users/login')
+    .send({
+      email: dummyUser[1].email,
+      password: dummyUser[1].password
+    })
+    .end((err, res) => {
+      adminToken = res.body.Token;
+      done();
+    });
+  });
+
+  describe('User Creation and Login', () => {
+    it('should create a user on /api/users POST', (done) => {
+      chai.request(app)
+      .post('/api/users')
+      .send({
+        firstName: dummyUser[3].firstName,
+        lastName: dummyUser[3].lastName,
+        userName: dummyUser[3].userName,
+        email: dummyUser[3].email,
+        password: dummyUser[3].password
+      })
+      .end((err, res) => {
+        res.should.have.status(201);
+        res.body.message.should.equal('User created succesfully');
+        res.body.firstName.should.equal(dummyUser[3].firstName);
+        done();
+      });
+    });
+    it('should create a unique user on /api/users POST', (done) => {
+      chai.request(app)
+      .post('/api/users')
+      .send({
+        firstName: dummyUser[1].firstName,
+        lastName: dummyUser[1].lastName,
+        userName: dummyUser[1].userName,
+        email: dummyUser[1].email,
+        password: dummyUser[1].password
+      })
+      .end((err, res) => {
+        res.should.have.status(400);
+        res.body[0].message.should.equal('email must be unique');
+        done();
+      });
+    });
+    it('should create a role for the user on /api/users POST', (done) => {
+      chai.request(app)
+      .post('/api/users')
+      .send({
```

Comment:
**Question** - Can we send just `dummyUser[2]` instead of copying over it's properties?
Like
```javascript
...
.send(dummyUser[2])
...
```